scheduler/device.go

```diff
@@ -76,14 +78,15 @@ func (d *deviceAllocator) AssignDevice(ask *structs.RequestedDevice) (out *struc
 					continue
 				}
 
-				totalWeight += a.Weight
+				totalWeight += math.Abs(a.Weight)
 
 				// Check if satisfied
 				if !checkAttributeAffinity(d.ctx, a.Operand, lVal, rVal) {
 					continue
 				}
 				choiceScore += a.Weight
 			}
+			matchedWeights += choiceScore
```

Comment:
Hmm this is over accounting. Shouldn't this be after: https://github.com/hashicorp/nomad/pull/4855/files#diff-2ffd1ab21e7ba9159b1cd9f8919596edR127